app/src/openxr/cpp/DeviceDelegateOpenXR.cpp

```diff
@@ -273,18 +273,71 @@ struct DeviceDelegateOpenXR::State {
 
   void InitializeSwapChainFormats() {
     uint32_t swapchainFormatCount;
-    CHECK_XRCMD(xrEnumerateSwapchainFormats(session, 0, &swapchainFormatCount, nullptr));
-    CHECK_MSG(swapchainFormatCount > 0, "OpenXR unexpected swapchainFormatCount");
+    RETURN_VOID_XR_FAILED(xrEnumerateSwapchainFormats(session, 0, &swapchainFormatCount, nullptr));
+    if (swapchainFormatCount > 0) {
+      VRB_ERROR("OpenXR unexpected swapchainFormatCount");
+    }
     swapchainFormats.resize(swapchainFormatCount, 0);
-    CHECK_XRCMD(xrEnumerateSwapchainFormats(session, (uint32_t)swapchainFormats.size(), &swapchainFormatCount,
+    RETURN_VOID_XR_FAILED(xrEnumerateSwapchainFormats(session, (uint32_t)swapchainFormats.size(), &swapchainFormatCount,
                                             swapchainFormats.data()));
     VRB_LOG("OpenXR Available color formats: %d", swapchainFormatCount);
   }
 
-  bool SupportsColorFormat(int64_t aColorFormat) {
+  int64_t GetColorFormat(std::string extension = "") {
     if (swapchainFormats.size() == 0) {
       InitializeSwapChainFormats();
     }
+
+    int64_t targetFormat;
+    int64_t fallbackFormat;
+    if (extension == ".ktx") {
+      switch (deviceType) {
+        case device::OculusGo:
+        case device::OculusQuest:
+        case device::OculusQuest2:
+          targetFormat = GL_COMPRESSED_SRGB8_ETC2;
```

Comment:
Could we perhaps join both switches clauses like this ? 

case device::OculusQuest2:
  targetFormat = extension == ".ktx" ? GL_COMPRESSED_SRGB8_ETC2 : GL_RGBA8 

Just an idea, what do you think ?

Reply:
Actually I prefer to keep the current way to have minimum code duplication of defining color format. After all ktx extensions and others are doomed to be different in color format as it’s compressed.